AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -349,17 +374,109 @@ - (void)createOrReplaceWithPartition:(NSString *)partition
+                          httpMethod:kMSHttpMethodGet
+                                body:nil
+                   additionalHeaders:additionalHeaders
+                   completionHandler:^(NSData *_Nullable data, NSHTTPURLResponse *_Nullable response, NSError *_Nullable cosmosDbError) {
+                     // If not OK.
+                     if (!data || [MSDataSourceError errorCodeFromError:cosmosDbError] != MSACDocumentSucceededErrorCode) {
+                       MSLogError([MSDataStore logTag], @"Not able to retrieve documents: %@", [cosmosDbError description]);
+                       MSDataSourceError *dataSourceCosmosDbError = [[MSDataSourceError alloc] initWithError:cosmosDbError];
+                       MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataSourceCosmosDbError];
+                       completionHandler(documents);
+                       return;
+                     }
+
+                     // Deserialize the list payload and try to get the array of documents.
+                     NSError *deserializeError;
+                     id jsonPayload = [NSJSONSerialization JSONObjectWithData:(NSData *)data options:0 error:&deserializeError];
+                     NSDictionary *jsonPayloadDict = nil;
+                     NSArray *jsonDocuments = nil;
+                     if (!deserializeError && jsonPayload && [(NSObject *)jsonPayload isKindOfClass:[NSDictionary class]]) {
+                       jsonPayloadDict = (NSDictionary *)jsonPayload;
+                     }
+                     if (jsonPayloadDict && [jsonPayloadDict objectForKey:@"Documents"] &&
+                         [(NSObject *)jsonPayloadDict[@"Documents"] isKindOfClass:[NSArray class]]) {
+                       jsonDocuments = jsonPayloadDict[@"Documents"];
+                     }
+                     if (!jsonDocuments) {
+                       if (!deserializeError)
+                         deserializeError = [[NSError alloc] initWithDomain:kMSACDataStoreErrorDomain
+                                                                       code:MSACDataStoreErrorJSONSerializationFailed
+                                                                   userInfo:@{NSLocalizedDescriptionKey : @"Can't deserialize documents"}];
+                       MSDataSourceError *dataSourceDeserializeError = [[MSDataSourceError alloc] initWithError:deserializeError];
+                       MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataSourceDeserializeError];
+                       completionHandler(documents);
+                       return;
+                     }
+
+                     // Parse the documents.
+                     NSMutableArray<MSDocumentWrapper *> *items = [NSMutableArray new];
+                     for (id document in jsonDocuments) {
+                       // Deserialize current document.
+                       // TODO: handle deserialization here.
```

Comment:
Hadn't we done this and the wrapper creation in the previous PR?  Not sure why they're TODO now

Reply:
Actually I'm not sure why this TODO is here at all, it seems like you're already doing deserialization here

Reply:
We rely on user's implementation. If the deserialization method is not present, this throws some exceptions. The user's implementation could fail too so I think we should always wrap any kind of user serialization/deserialization in a try/catch block. It's not only list API it's all the read/writes, I will add a task on the board, this PR is getting big.

Reply:
Actually I was referring to the TODO few lines below. You are correct that one should not be there (artifacts of my conflicts resolution skills :)).

Fixed (my previous comment still holds but for the next TODO).